rootfs/etc/nginx/lua/configuration.lua

```diff
@@ -27,13 +29,67 @@ local function fetch_request_body()
   return body
 end
 
+-- Returns the certificate and key for a given host
+function _M.get_cert_key(hostname)
+  return configuration_data:get(hostname)
+end
+
+local function handle_cert_request()
+  if ngx.var.request_method ~= "POST" then
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    ngx.print("Only POST requests are allowed!")
+    return
+  end
+
+  local raw_certs = fetch_request_body()
+
+  local ok, certs = pcall(json.decode, raw_certs)
+  if not ok then
+    ngx.log(ngx.ERR,  "could not parse certificate: " .. tostring(certs))
+    return
+  end
+
+  if not certs then
+    ngx.log(ngx.ERR, "certificate dynamic-configuration: unable to read valid request body")
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    return
+  end
+
+  local err_buf = {}
+  -- Update certificates and private keys for each host
+  for _, cert in pairs(certs) do
```

Comment:
for an array it's recommended to use `ipairs`.

`pairs` for when it is a dictionary